### ISSUE TITLE
Touching home when already home causes the app close itself

### ISSUE THREAD

**Issue by fluktuid** [k0shk0sh/FastHub#411]
**App Version: 2.0.1**

**OS Version: 25**

**Model: OnePlus-ONEPLUS A3003**

 If I'm at home and press home on the navigationdrawer the app closes

**Comment by kutsan** [comment]
Thanks for pointing out. Hopefully, will be fixed the following releases.

**Comment by arkon** [comment]
On the same topic, the other selected items cause an extra instance of the activity to be added to the stack.

**Comment by k0shk0sh** [comment]
@arkon it will definitely start new instance unless if you want me to restrict you to have one thing opened at a time. I'm with shouldn't be clicked again if it was selected 😁

**Comment by camillo777** [comment]
If FastHub is started from Google play Store page (open button), pressing home from FastHub menu will go back to play Store page; I think it is the same bug.

**Comment by k0shk0sh** [comment]
@camillo777 yep the same

**Comment by k0shk0sh** [comment]
@camillo777 the life cycle will never be simple in here it's complicated more than you think, you are at a repo, went to an issue, you are expecting that once you press back to go where you left, but then you opened a link in the issue comments that takes you to another issue, would you want to go back to previous issue or home? Of course the previous screen where you left, then in this case, having a `singleTop` activity won't work. Only `Home` activity is a `singleTop` activity that can't be recreated, therefore once you click home again, that activity is destroyed and therefore you are out of FastHub.

**Comment by camillo777** [comment]
@k0shk0sh but aren't you using fragments?

**Comment by k0shk0sh** [comment]
@camillo777 couldn't get your point?

**Comment by camillo777** [comment]
@k0shk0sh maybe I do not understand correctly, but using fragments on Android allow to have one single activity in the application, changing only the content. You just have to change displayed fragments and manage activity life cycle only once. 

This is the Google best practice for Android development. 

But maybe I  missing something.

**Comment by k0shk0sh** [comment]
@camillo777  `Fragments` are being used heavily in __FastHub__ but that doesn't mean that I'm using a single activity approach, it's not applicable for such project.
